grr/server/grr_response_server/gui/ui/components/flow_details/plugins/osquery_details.ts

```diff
@@ -15,58 +15,63 @@ import { isNonNull } from '@app/lib/preconditions';
   styleUrls: ['./osquery_details.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class OsqueryDetails extends Plugin implements OnDestroy {
-  readonly unsubscribe$ = new Subject<void>();
-
+export class OsqueryDetails extends Plugin {
   readonly flowRunning$ = this.flagByState(FlowState.RUNNING);
   readonly flowCompleted$ = this.flagByState(FlowState.FINISHED);
   readonly flowError$ = this.flagByState(FlowState.ERROR);
 
   readonly args$: Observable<OsqueryArgs> = this.flowListEntry$.pipe(
-      map((flowListEntry) => flowListEntry.flow.args as OsqueryArgs)
+      map(flowListEntry => flowListEntry.flow.args as OsqueryArgs),
   );
 
-  readonly osqueryResult$: Observable<OsqueryResult> = this.flowListEntry$.pipe(
-    flatMap((listEntry) => listEntry.resultSets),
-    flatMap((singleResultSet) => singleResultSet?.items),
+  readonly osqueryResults$: Observable<OsqueryResult> = this.flowListEntry$.pipe(
+    flatMap(listEntry => listEntry.resultSets),
+    flatMap(singleResultSet => singleResultSet?.items),
     filter(isNonNull),
-    map((singleItem) => singleItem.payload as OsqueryResult),
+    map(singleItem => singleItem.payload as OsqueryResult),
+  );
+
+  readonly resultsTable$ = this.osqueryResults$.pipe(
+    map(result => result?.table),
   );
 
-  readonly resultTable$ = this.osqueryResult$.pipe(
-    map((result) => result?.table),
+  readonly resultsStderr$ = this.osqueryResults$.pipe(
+    map(result => result?.stderr),
   );
 
-  readonly resultStderr$ = this.osqueryResult$.pipe(
-    map((result) => result?.stderr)
+  readonly osqueryProgress$: Observable<OsqueryProgress> = this.flowListEntry$.pipe(
+    map(listEntry => listEntry.flow.progress as OsqueryProgress),
+    filter(isNonNull),
+  );
+
+  readonly progressTable$ = this.osqueryProgress$.pipe(
+    map(progress => progress.partialTable),
+  );
+
+  readonly additionalRowsAvailable$ = this.osqueryProgress$.pipe(
+    map(progress => {
+      const progressTableLength = progress.partialTable?.rows?.length;
+      const fullTableLength = progress.totalRowsCount;
+
+      if (isNonNull(progressTableLength) && isNonNull(fullTableLength)) {
+        return BigInt(fullTableLength) - BigInt(progressTableLength);
```

Comment:
BigInt is only needed if your numbers are bigger than `Number.MAX_SAFE_INT` (~9P). In this case it is safe to parse the decimal string `progress.totalRowsCount` as `Number` instead of `BigInt`.